dao/src/main/java/com/carhouse/dao/impl/CarSaleDaoImpl.java

```diff
@@ -122,14 +134,24 @@ public CarSale getCarSale(final int carSaleId) {
      * Adds car sale advertisement to the end of list in database
      *
      * @param carSale the car sale
+     * @param file    the image file
      * @return car sale id
+     * @throws FileSystemException the file system exception when writing file
      */
     @Override
-    public Integer addCarSale(final CarSale carSale) {
+    public Integer addCarSale(final CarSale carSale, final MultipartFile file) throws FileSystemException {
         LOGGER.debug("method addCarSale with parameter: [{}]", carSale);
         KeyHolder keyHolder = new GeneratedKeyHolder();
+        if (!file.isEmpty()) {
+            carSale.setImageName(UUID.randomUUID().toString());
+        } else {
+            carSale.setImageName("default");
+        }
         namedParameterJdbcTemplate.update(ADD_CAR_SALE_SQL, parameterSource.getCarSaleParameters(carSale), keyHolder,
                 new String[]{"car_sale_id"});
+        if (!file.isEmpty()) {
```

Comment:
line 145 has the same check.
Why this action couldn't been done there?>

Reply:
Because I first try to add carSale to the database and, if successful, I write the image to the Apache server